File: C Projects/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");

    double lett = count_letters(text);
    double word = count_words(text);
    double sent = count_sentences(text);

    //Coleman-Liau index
    double L = lett / word * 100;
    double S = sent / word * 100;
    double index = 0.0588 * L - 0.296 * S - 15.8;
    int rounded = round(index);

    //Print the grade level
    if (rounded < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (rounded >= 1 && rounded <= 15)
    {
        printf("Grade %i\n", rounded);
    }
    else
    {
        printf("Grade 16+\n");
    }
}

//Count letters
int count_letters(string text)
{
    int lett = 0;
    for (int i = 0, l = strlen(text); i < l; i++)
    {
        if (isalpha(text[i]))
        {
            lett += 1;
        }
        else
        {
            lett += 0;
        }
    }
    return lett;
}

//Count words
int count_words(string text)
{
    int word = 0;
    for (int i = 0, l = strlen(text); i < l; i++)
    {
        if (isspace(text[i]))
        {
            word += 1;
        }
        else
        {
            word += 0;
        }
    }
    word += 1;
    return word;
}

//Check sentences
int count_sentences(string text)
{
    int sent = 0;
    for (int i = 0, l = strlen(text); i < l; i++)
    {
        if (text[i] == 46 || text[i] == 33 || text[i] == 63)
        {
            sent += 1;
        }
        else
        {
            sent += 0;
        }
    }
    return sent;
}

File: Python Projects/readability Python-ver/readability.py
# TODO
def main():
    text = input("Text: ")
    l = count_letters(text) / count_words(text) * 100
    s = count_sentences(text) / count_words(text) * 100
    cl_index = round(0.0588 * l - 0.296 * s - 15.8)
    print_grade(cl_index)


def count_letters(text):
    counter = 0
    for char in text:
        if char.isalpha():
            counter += 1
    return counter


def count_words(text):
    counter = 1
    for space in text:
        if space.isspace():
            counter += 1
    return counter


def count_sentences(text):
    counter = 0
    for punc in text:
        if punc in [".", "?", "!"]:
            counter += 1
    return counter


def print_grade(n):
    if n < 1:
        print("Before Grade 1")
    elif 1 <= n <= 15:
        print(f"Grade {n}")
    else:
        print("Grade 16+")


main()
